module/VuFind/src/VuFind/Db/Entity/Record.php

```diff
@@ -109,4 +110,99 @@ class Record implements EntityInterface
      * )
      */
     protected $updated = '2000-01-01 00:00:00';
+
+    /**
+     * Record Id setter
+     *
+     * @param ?string $id Record Id.
+     *
+     * @return Record
+     */
+    public function setRecordId(?string $id): Record
+    {
+        $this->recordId = $id;
+        return $this;
+    }
+
+    /**
+     * Get record id.
+     *
+     * @return ?string
+     */
+    public function getRecordId(): ?string
+    {
+        return $this->recordId;
+    }
+
+    /**
+     * Record source setter
+     *
+     * @param ?string $source Record source.
+     *
+     * @return Record
+     */
+    public function setSource(?string $source): Record
+    {
+        $this->source = $source;
+        return $this;
+    }
+
+    /**
+     * Get record source.
+     *
+     * @return ?string
+     */
+    public function getSource(): ?string
+    {
+        return $this->source;
+    }
+
+    /**
+     * Record data setter
+     *
+     * @param ?string $data Record data.
+     *
+     * @return Record
+     */
+    public function setData(?string $data): Record
+    {
+        $this->data = $data;
+        return $this;
+    }
+
+    /**
+     * Get record data.
+     *
+     * @return ?string
+     */
+    public function getData(): ?string
+    {
+        return $this->data;
+    }
+
+    /**
+     * Record version setter
+     *
+     * @param string $version Record version.
+     *
+     * @return Record
+     */
+    public function setVersion(string $version): Record
```

Comment:
Is there a reason you don't have getters for Version and Updated?

Reply:
They are nowhere used, just written the setters and getters that are used. Do you want to add them?

Reply:
It seems worth adding them; I suspect we'll find them useful for testing if nothing else.